src/Converter.cc

```diff
@@ -229,13 +240,304 @@ void Converter::ConvertImpl(tinyxml2::XMLElement *_elem,
+            "Error: Pose attribute 'relative_to' does not start with " +
+            newModelName);
+
+          poseRelTo = poseRelTo.substr(_newNameIdx);
+          poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+        }
+      }
+    }  // link
+
+    else if (elemName == "joint")
+    {
+      // strip new model prefix from //joint/parent
+      tinyxml2::XMLElement *e = elem->FirstChildElement("parent");
+      std::string eText = e->GetText();
+
+      SDF_ASSERT(eText.compare(0, _newNameIdx - 2, newModelName) == 0,
+        "Error: Joint's <parent> value does not start with " + newModelName);
+
+      e->SetText(eText.substr(_newNameIdx).c_str());
+
+      // strip new model prefix from //joint/child
+      e = elem->FirstChildElement("child");
+      eText = e->GetText();
+
+      SDF_ASSERT(eText.compare(0, _newNameIdx - 2, newModelName) == 0,
+        "Error: Joint's <child> value does not start with " + newModelName);
+
+      e->SetText(std::string(e->GetText()).substr(_newNameIdx).c_str());
+
+      // strip new model prefix from //xyz/@expressed_in
+      std::string axisStr = "axis";
+      while (true)
+      {
+        tinyxml2::XMLElement *axisElem =
+            elem->FirstChildElement(axisStr.c_str());
+        if (axisElem != nullptr)
+        {
+          if (axisElem->FirstChildElement("xyz")->Attribute("expressed_in"))
+          {
+            std::string expressIn =
+                axisElem->FirstChildElement("xyz")->Attribute("expressed_in");
+
+            SDF_ASSERT(expressIn.compare(0, _newNameIdx - 2, newModelName) == 0,
+              "Error: <xyz>'s attribute 'expressed_in' does not start with " +
+              newModelName);
+
+            expressIn = expressIn.substr(_newNameIdx);
+
+            axisElem->FirstChildElement("xyz")
+                    ->SetAttribute("expressed_in", expressIn.c_str());
+          }
+        }
+
+        if (axisStr == "axis2") break;
+
+        axisStr += "2";
+      }
+
+      // strip new model prefix from all //joint/sensor/pose/@relative_to
+      for (e = elem->FirstChildElement("sensor");
+          e;
+          e = e->NextSiblingElement("sensor"))
+      {
+        poseElem = e->FirstChildElement("pose");
+        if (poseElem != nullptr)
+        {
+          std::string poseRelTo = poseElem->Attribute("relative_to");
+
+          SDF_ASSERT(poseRelTo.compare(0, _newNameIdx - 2, newModelName) == 0,
+            "Error: Pose attribute 'relative_to' does not start with " +
+            newModelName);
+
+          poseRelTo = poseRelTo.substr(_newNameIdx);
+          poseElem->SetAttribute("relative_to", poseRelTo.c_str());
+        }
+      }
+    }  // joint
+
+    else if (elemName == "gripper")
+    {
+      bool hasPrefix = true;
+
+      // strip prefix from all /model/gripper/gripper_link
+      tinyxml2::XMLElement *e = elem->FirstChildElement("gripper_link");
+      std::string eText;
+      while (e)
+      {
+        eText = e->GetText();
```

Comment:
Check if `e->GetText()` is nullptr, here and for `palm_link`.

Reply:
dce6828593004010ad58c44ab45f8d0b7bf462be